src/server.c

```diff
@@ -3042,13 +3042,36 @@ int listenToPort(int port, int *fds, int *count) {
                 continue;
             return C_ERR;
         }
-        anetNonBlock(NULL,fds[*count]);
-        anetCloexec(fds[*count]);
-        (*count)++;
+        anetNonBlock(NULL,sfd->fd[sfd->count]);
+        anetCloexec(sfd->fd[sfd->count]);
+        sfd->count++;
     }
     return C_OK;
 }
 
+void closeSocketListeners(socketFds *sfd) {
+    int j;
+
+    for (j = 0; j < sfd->count; j++) {
+        aeDeleteFileEvent(server.el, sfd->fd[j], AE_READABLE);
+        close(sfd->fd[j]);
+    }
+
+    sfd->count = 0;
+}
+
+void createSocketAcceptHandler(socketFds *sfd, aeFileProc *accept_handler) {
+    int j;
+
+    /* Create an event handler for accepting new connections in TCP and TLS
+     * domain sockets. */
+    for (j = 0; j < sfd->count; j++) {
+        if (aeCreateFileEvent(server.el, sfd->fd[j], AE_READABLE, accept_handler,NULL) == AE_ERR) {
+            serverPanic("Unrecoverable error creating file event.");
```

Comment:
Consider simply cleaning up and returning an error, then maybe `changeListenPort` could then handle errors more gracefully rather than panic.

Reply:
fixed